cleanlab_studio/utils/databricks_utils.py

```diff
@@ -0,0 +1,135 @@
+import os
+import zipfile
+from datetime import datetime
+from pathlib import Path
+
+
+def dbfs_to_posix_path(dbfs_path: str) -> str:
+    """
+    Converts a DBFS path to a POSIX path.
+
+    Args:
+        dbfs_path: The DBFS path to convert.
+
+    Returns:
+        The POSIX path.
+    """
+    return f"/dbfs{dbfs_path[5:]}"
+
+
+def get_databricks_imageset_df_image_col(df) -> str:
+    from pyspark.sql.types import StructField, StructType
+    from pyspark.sql.types import StringType, IntegerType, BinaryType
+
+    # check for image column
+    required_image_fields = [
+        StructField("origin", StringType(), True),
+        StructField("height", IntegerType(), True),
+        StructField("width", IntegerType(), True),
+        StructField("nChannels", IntegerType(), True),
+        StructField("mode", IntegerType(), True),
+        StructField("data", BinaryType(), True),
+    ]
+
+    struct_col_schemas = [f for f in df.schema.fields if isinstance(f.dataType, StructType)]
+
+    for s in struct_col_schemas:
+        s_fields = s.dataType.fields
+        if all([f in s_fields for f in required_image_fields]):
+            return s.name
+
+    return None
+
+
+def create_path_based_imageset_archive(folder_path, archive_name=None) -> str:
+    """
+    Archives an imageset stored on Databricks which can then be uploaded Cleanlab Studio.
+    The imageset folder should match the layout described in the Cleanlab Studio documentations.
+    https://help.cleanlab.ai/guide/concepts/datasets/#image
+
+    Args:
+        folder_path: The POSIX-style path to your imageset.
+        archive_name: A name you choose for the imageset archive.
+
+    Returns:
+        The relative path to the archived imageset.
+    """
+    folder_name = Path(folder_path).resolve(strict=False).name
+    archive_name = (
+        archive_name
+        if archive_name is not None
+        else f'{folder_name}_archive_{datetime.now().strftime("%Y-%d-%m_%H:%M:%S")}'
+    )
+    output_filename = f"{archive_name}.zip"
+
+    # Create a ZipFile object in write mode
+    with zipfile.ZipFile(output_filename, "w", zipfile.ZIP_DEFLATED) as zipf:
+        # Walk through the directory
+        for root, _, files in os.walk(folder_path):
```

Comment:
I believe `pathlib.Path().walk()` should be roughly equivalent to `os.walk()` https://docs.python.org/3/library/pathlib.html#pathlib.Path.walk

Reply:
Doesn't seem to be included in Databricks' default env.
![Screen Shot 2024-01-25 at 10 57 58 PM](https://github.com/cleanlab/cleanlab-studio/assets/19893923/8409ceea-77c4-4608-a030-a6e15a1b7905)

Reply:
Oh I think it's new in python 3.12. Probably best not to use then since we don't want to force people to be on latest python version

Reply:
It should be possible to do this with other pathlib methods, but since we're low on time this is fine